### DISCUSSION TITLE
How to add support for new sensor?

### DISCUSSION THREAD

**Discussion by nickagian** [custom-components/ble_monitor#337] [General]
Hi all,

I'm interested in extending the supported sensors with custom designed hw. I was thus wondering if you can help me with this. What are roughly the steps I need to follow for this? I am obviously not beginner in python, but still it would be nice if I didn't start completely in the dark :-)

Nikos

**Comment by Ernst79** [comment]
You have two options, 
1. make a fake Xiaomi sensor that follows the format of the Xiaomi MiBeacon format
2. create your own BLE format and and I will help you out with adding it in BLE monitor

In case you want to follow 1, check [this issue](https://github.com/custom-components/ble_monitor/issues/222) for someone who did something similar
In case you want to follow 2, post a BLE message of your sensor here, (you can get it with a HCIDUMP.) 

also check [this page](https://github.com/custom-components/ble_monitor/blob/master/protocol.md) for some inspiration on how a message can look like. If you want your own format, use a different UUID. There is a [list on internet from the bluetooth foundation](https://btprodspecificationrefs.blob.core.windows.net/assigned-values/16-bit%20UUID%20Numbers%20Document.pdf) with all assigned UUIDs.
